collector/hive/examples/compose/hive-server/Dockerfile

```diff
@@ -0,0 +1,27 @@
+FROM curlimages/curl:7.82.0 as curler
+ARG JMX_PROM_VERSION=0.20.0
+USER root
+
+RUN curl -L \
+    --output /jmx_prometheus_javaagent.jar \
+    "https://repo.maven.apache.org/maven2/io/prometheus/jmx/jmx_prometheus_javaagent/${JMX_PROM_VERSION}/jmx_prometheus_javaagent-${JMX_PROM_VERSION}.jar"
+
+
+# Using the base Hive image
+FROM bde2020/hive:2.3.2-postgresql-metastore
+
+# Copy the jmx_prometheus_javaagent jar and jmx_exporter.yaml config
+COPY --from=curler /jmx_prometheus_javaagent.jar /opt/hive/
+COPY jmx_exporter.yaml /opt/hive/conf/
+
+# Append script to hive-env.sh
```

Comment:
One way to clean this up would be put this as a script in the build context and just append the call to that script in the sh. Take 30 minutes max to clean that up for readability. If it takes longer we have people who specialize in that for you ;)